keyboards/handwired/ferris/ferris.c

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2013 Oleg Kostyuk <cub.uanic@gmail.com>
+          2020 Pierre Chevalier <pierrechevalier83@gmail.com>
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include QMK_KEYBOARD_H
+
+bool         i2c_initialized = 0;
+i2c_status_t mcp23017_status = I2C_ADDR;
+
+void matrix_init_kb(void) {
+    // set unused pins on mcu as input with internal pull-up enabled
+    pin_t unused_pins_mcu[UNUSED_MCU] = UNUSED_PINS_MCU;
+    for (int pin_index = 0; pin_index < UNUSED_MCU; pin_index++) {
+        pin_t pin = unused_pins_mcu[pin_index];
+        setPinInputHigh(pin);
+    }
+
+    matrix_init_user();
+}
+
```

Comment:
Do we still need this, given that we're now using `CUSTOM_MATRIX = lite`?
Could they be folded into `matrix_init_custom()`?

Reply:
Uhm. I followed the documentation from CUSTOM_MATRIX to decide what I was going to remove, although this code was heavily inspired by ergodox and gergoplex, so I may have inherited some older conventions from there without fully understanding. I'll play around with removing more code and see how much still works without bricking my Ferris :)

Reply:
Yes, it can go :)